src/ar_serial_client.c

```diff
@@ -0,0 +1,277 @@
+}
+
+
+//-------------------------------------------------------------- 
+// MAIN
+//--------------------------------------------------------------
+
+int main(int argc, char* argv[]){
+	// serial variables
+	int serial_read_port = 0;
+	int count = 0;
+
+	// socket variables
+	int sock_fd = 0;
+	int tcp_write_port = 0;
+	int message;
+	struct sockaddr_in server_address;
+	struct hostent *server;
+	char* host;
+	char buff[BUFFER];	
+
+	// command line check
+	if(argc != 3){
+		fprintf(stderr, "usage: %s <hostname> <port>\n", argv[0]);
+		exit(0);
+	}
+
+//-------------------------------------------------------------- 
+// SERIAL SETUP BEGIN
+//--------------------------------------------------------------
+	// try to open the port and check for errors	
+	serial_read_port = open(OPEN_PORT, O_RDWR);
+	// attempt to open port 20 times before quiting
+	while(serial_read_port <= 0 && count < 20){
+		count++;
+		// error messages
+		display_non_errno_error("Retrying serial port connection...");
+		if(DEBUG_MODE){
+			printf("Connections attempted: %d\n", count);
+			display_errno_error(errno, "open");
+		}
+	}
+
+	// initialize tty of type termios and read in current port data
+	struct termios tty;
+	if(tcgetattr(serial_read_port, &tty) !=0){
+		display_errno_error(errno, "tcgetattr");
+		if(!DEBUG_MODE){
+			return 1;
+		}
+	}
+
+	// configure termios port
+	set_termios_flags(&tty);
+	// configure baud rate in and out
+	cfsetispeed(&tty, B9600);
+	cfsetospeed(&tty, B9600);
+
+	// save all flags set above
+	if(tcsetattr(serial_read_port, TCSANOW, &tty) != 0){
+		display_errno_error(errno, "tcsetattr");
+		if(!DEBUG_MODE){
+			return 1;
+		}
+	}
+//--------------------------------------------------------------
+// SERIAL SETUP END
+//--------------------------------------------------------------
+
+//-------------------------------------------------------------- 
+// TCP SETUP BEGIN
+//--------------------------------------------------------------
+	// assign hostname and port from cli
+	host = argv[1];
+	tcp_write_port = atoi(argv[2]);
+
+	// socket setup
+	sock_fd = socket(AF_INET, SOCK_STREAM, 0);
+	if(sock_fd < 0){
+		display_perror_error("Error creating socket");
+	}
+
+	// server setup
+	server = gethostbyname(host);
+	if(server == NULL){
+		fprintf(stderr, "Cannot find host: %s\n", host);
+		exit(0);
+	}
+	bzero((char*) &server_address, sizeof(server_address));
+	server_address.sin_family = AF_INET;
+	bcopy((char*) server->h_addr, (char*) &server_address.sin_addr.s_addr, server->h_length);
+	server_address.sin_port = htons(tcp_write_port);
+
+	// server connect
+	if(connect(sock_fd, &server_address, sizeof(server_address)) < 0 ){
+		display_perror_error("Error connecting to server");
+	}
+//-------------------------------------------------------------- 
+// TCP SETUP END
+//--------------------------------------------------------------
+
+	while(1){
+		bzero(buff, BUFFER);
+		read_serial_data(serial_read_port, buff);
+
```

Comment:
maybe i am stupid but where is the code to write to the server?

Reply:
also i dunno the behavior but if you send the whole buffer after it has been bzero'd will that pad it with 0's? i just want to make sure we are always sending the name number of bits to the server